trie/database.go

```diff
@@ -126,11 +135,11 @@ func NewDatabase(diskdb ethdb.Database, config *Config) *Database {
 	}
 	var preimages *preimageStore
 	if config.Preimages {
-		preimages = newPreimageStore(diskdb)
+		preimages = newPreimageStore(statediskdb)
```

Comment:
Why store preimages to statediskdb? Although currently its disable in BSC.

Reply:
statediskdb indicates  the diskdb of trie database here .  It is equal to chaindb if no separate db has set.    preimages store in the same diskdb of trie db.